File: src/utils/redis_client.ts
import redis from 'redis';


export default class RedisClient {
  private client: redis.RedisClient;

  constructor(port, host) {
    this.client = redis.createClient(port, host);
  }

  writeToRedis = (key: string, value: string, ttl = 60): void => {
    this.client.set(key, value, 'EX', ttl);
  }

  readFromRedis = async (key: string) => {
    this.client.get(key, () => {

    });

    await this.client.get(key, (error, reply) => {
      if (error) {
        throw new Error(error.message);
      }

      console.log(reply);
    });
  }
}

File: app/utils/redis_client.py
import redis


class RedisClient:
    def __init__(self, port: int, host: str):
        self.client = redis.Redis(host=host, port=port, db=0)

    def writeToRedis(self, key: str, value: str) -> None:
        self.client.set(key, value)

    def readFromCache(self, key: str) -> str:
        return self.client.get(key)
